cli/lib/nf3.mjs

```diff
@@ -95,7 +86,10 @@ class Nf3 {
     if (typeof this.ethereumSigningKey === 'string') {
       this.ethereumAddress = await this.getAccounts();
     }
-    return this.subscribeToIncomingViewingKeys();
+    // set zkp keys from mnemonic if provided
+    if (typeof mnemonic !== 'undefined') {
+      this.setzkpKeysFromMnemonic(mnemonic, 0);
```

Comment:
I think you need to await `this.setzkpKeysFromMnemonic` or return it as a promise, otherwise `init` will return before the `setzkpKeysFromMnemonic` has finished.  Thus code that is awaiting `init` may not run correctly.

Reply:
You're right.